includes/sanitizers/class-amp-form-sanitizer.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Class AMP_Form_Sanitizer.
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Form_Sanitizer
+ *
+ * Strips and corrects attributes in forms.
+ */
+class AMP_Form_Sanitizer extends AMP_Base_Sanitizer {
+
+	/**
+	 * Tag.
+	 *
+	 * @var string HTML <form> tag to identify and process.
+	 *
+	 * @since 0.2
+	 */
+	public static $tag = 'form';
+
+	/**
+	 * Sanitize the <img> elements from the HTML contained in this instance's DOMDocument.
+	 *
+	 * @since 0.2
+	 */
+	public function sanitize() {
+
+		/**
+		 * Node list.
+		 *
+		 * @var DOMNodeList $node
+		 */
+		$nodes     = $this->dom->getElementsByTagName( self::$tag );
+		$num_nodes = $nodes->length;
+
+		if ( 0 === $num_nodes ) {
+			return;
+		}
+
+		for ( $i = $num_nodes - 1; $i >= 0; $i-- ) {
+			$node = $nodes->item( $i );
+			if ( ! $node instanceof DOMElement ) {
+				continue;
+			}
+
+			if ( ! $node->hasAttribute( 'action' ) || '' === $node->getAttribute( 'action' ) ) {
```

Comment:
Per the spec: https://www.ampproject.org/docs/reference/components/amp-form#target

> This attribute is required for `method=GET`. For `method=POST`, the `action` attribute is invalid, use `action-xhr` instead.

Is the second part of this conditional intended to check if `'get' === $node->getAttribute( 'method' )`? Since the default method is `get` then I think there should be something above that does:

```php
$method = 'get';
if ( $node->hasAttribute( 'method' ) ) {
    $method = strtolower( $node->getAttribute( 'method' ) );
}
```

And then use the `$method` variable going forward.